pkg/controllers/state/cluster.go

```diff
@@ -391,6 +393,7 @@ func (c *Cluster) MarkPodSchedulingDecisions(ctx context.Context, podErrors map[
 			}
 		}
 		c.podHealthyNodePoolScheduledTime.Delete(nn)
+		c.podToNodeClaim.Delete(nn)
```

Comment:
Do we not delete from this map when we are clearingDecisions? I think this map could grow unbounded